xspress3App/src/xspress3Epics.cpp

```diff
@@ -1142,24 +1180,35 @@ asynStatus Xspress3::setTriggerMode(int mode, int invert_f0, int invert_veto, in
     asynPrint(this->pasynUserSelf, ASYN_TRACE_FLOW, "%s Set Trigger Mode.\n", functionName);
     getIntegerParam(xsp3NumCardsParam, &xsp3_num_cards);
     for (int card=0; card<xsp3_num_cards && status == asynSuccess; card++) {
-        if ( card == 0 ) {
-			if (mode == 7) {
-			mode = 1; // Refactor to 1 to continue to use the internal trigger generator but set up so won't continue until histogram_continue is called
-			}
-			// probably want to arm the histogram here
-            status = mapTriggerMode(mode, invert_f0, invert_veto, debounce, &xsp3_trigger_mode);
-        	}
-
-        else
-        {
-            status = mapTriggerMode(mbboTriggerTTLVETO_, invert_f0, 0, debounce, &xsp3_trigger_mode);
-        }
-
-        int xsp3_status = xsp3->set_glob_timeA(xsp3_handle_, card, xsp3_trigger_mode);
-        if (xsp3_status != XSP3_OK) {
-            checkStatus(xsp3_status, "xsp3_set_glob_timeA", functionName);
-            status = asynError;
+      if ( card == 0 ) {
+        if (mode == mbboTriggerSOFTINTERNAL_ || mode == mbboTriggerTTLINTERNAL_)  {
+          mode = 1; // Refactor to 1 to continue to use the internal trigger generator but set up so won't continue until histogram_continue is called
         }
+        // probably want to arm the histogram here
+          status = mapTriggerMode(mode, invert_f0, invert_veto, debounce, &xsp3_trigger_mode);
+      }
+      else
+      {
+          status = mapTriggerMode(mbboTriggerTTLVETO_, invert_f0, 0, debounce, &xsp3_trigger_mode);
+          // status = mapTriggerMode(mode, invert_f0, 0, debounce, &xsp3_trigger_mode);
+          // xsp3_trigger_mode = 0;
+      }
+      if (xsp3_is_xsp3m_plus(0) == 1) {
+        xsp3_trigger_mode = xsp3_trigger_mode | XSP3_GLOB_TIMA_FROM_RADIAL;
+      }
+      int xsp3_status = xsp3->set_glob_timeA(xsp3_handle_, card, xsp3_trigger_mode);
+      if (xsp3_status != XSP3_OK) {
+          checkStatus(xsp3_status, "xsp3_set_glob_timeA", functionName);
+          status = asynError;
+      }
+    // TODO Is this needed for Mk2?
+    // u_int32_t actual_trigger_mode;
+    // xsp3_get_glob_timeA(xsp3_handle_, card, &actual_trigger_mode);
+    // }
+    // int xsp3_status = xsp3->set_sync_mode(xsp3_handle_, XSP3_SYNC_MIDPLANE, 0 , 0);
+    // if (xsp3_status != XSP3_OK) {
+    //   checkStatus(xsp3_status, "xsp3_set_sync_mode", functionName);
+    //   status = asynError;
     }
```

Comment:
Remove if this is not actually needed for Mk2? Or check if we do need this logic to synchronise the cards.